homeassistant/components/bluetooth/manager.py

```diff
@@ -653,9 +659,13 @@ def _unregister_scanner() -> None:
             self._advertisement_tracker.async_remove_source(scanner.source)
             scanners.remove(scanner)
             del self._sources[scanner.source]
+            if connection_slots:
```

Comment:
I don't know if this matters, but you're not differentiating between `None` and `0`

Reply:
It shouldn't matter as we can never have an adapter with 0 connection slots (although with the quality of some of the adapters on the market they might as well effectively have 0 😉 ).